drivers/cc110x/cc110x_netdev.c

```diff
@@ -0,0 +1,610 @@
+    }
+}
+
+static int check_config(cc110x_t *dev)
+{
+    char buf[CC110X_CONF_SIZE];
+
+    /* Verify content of main config registers */
+    cc110x_burst_read(dev, CC110X_CONF_START, buf, sizeof(buf));
+    if (memcmp(buf, cc110x_conf, sizeof(buf))) {
+        DEBUG("[cc110x] ERROR: Verification of main config registers failed\n");
+        return -1;
+    }
+
+    /* Verify content of "magic number" config registers */
+    cc110x_burst_read(dev, CC110X_REG_TEST2, buf,
+                      sizeof(cc110x_magic_registers));
+    if (memcmp(buf, cc110x_magic_registers, sizeof(cc110x_magic_registers))) {
+        DEBUG("[cc110x] ERROR: Verification of \"magic\" registers failed\n");
+        return -1;
+    }
+
+    /* Verify content of PA_TABLE */
+    cc110x_burst_read(dev, CC110X_MULTIREG_PATABLE, buf, CC110X_PATABLE_LEN);
+    if (memcmp(buf, dev->params.patable->data, CC110X_PATABLE_LEN)) {
+        DEBUG("[cc110x] ERROR: Verification of PA_TABLE failed\n");
+        return -1;
+    }
+
+    DEBUG("[cc110x] Content of configuration registers verified\n");
+    return 0;
+}
+
+static int check_gdo_pins(cc110x_t *dev)
+{
+    /* GPIOs connected to GDOs are not yet configured, so we do this now */
+    if (gpio_init(dev->params.gdo2, GPIO_IN)) {
+        DEBUG("[cc110x] Configuring GDO2 failed");
+        return -1;
+    }
+
+    if (gpio_init(dev->params.gdo0, GPIO_IN)) {
+        DEBUG("[cc110x] Configuring GDO0 failed");
+        return -1;
+    }
+
+    /* Validate that GDO2 responds to configuration updates */
+    cc110x_write(dev, CC110X_REG_IOCFG2, CC110X_GDO_CONSTANT_HIGH);
+    if (!gpio_read(dev->params.gdo2)) {
+        DEBUG("[cc110x] GDO2 does not respond (check wiring!)\n");
+        return -1;
+    }
+
+    cc110x_write(dev, CC110X_REG_IOCFG2, CC110X_GDO_CONSTANT_LOW);
+    if (gpio_read(dev->params.gdo2)) {
+        DEBUG("[cc110x] GDO2 does not respond (check wiring!)\n");
+        return -1;
+    }
+
+    /* Validate that GDO0 responds to configuration updates */
+    cc110x_write(dev, CC110X_REG_IOCFG0, CC110X_GDO_CONSTANT_HIGH);
+    if (!gpio_read(dev->params.gdo0)) {
+        DEBUG("[cc110x] GDO0 does not respond (check wiring!)\n");
+        return -1;
+    }
+
+    cc110x_write(dev, CC110X_REG_IOCFG0, CC110X_GDO_CONSTANT_LOW);
+    if (gpio_read(dev->params.gdo0)) {
+        DEBUG("[cc110x] GDO0 does not respond (check wiring!)\n");
+        return -1;
+    }
+
+    /* Restore default GDO2 & GDO0 config */
+    cc110x_write(dev, CC110X_REG_IOCFG2, cc110x_conf[CC110X_REG_IOCFG2]);
+    cc110x_write(dev, CC110X_REG_IOCFG0, cc110x_conf[CC110X_REG_IOCFG0]);
+
+    return 0;
+}
+
+static int cc110x_init(netdev_t *netdev)
+{
+    cc110x_t *dev = (cc110x_t *)netdev;
+    /* Use locked mutex to block thread on TX and un-block from ISR */
+    mutex_init(&dev->isr_signal);
+    mutex_lock(&dev->isr_signal);
+
+    /* Make sure the crystal is stable and the chip ready */
+    if (cc110x_power_on(dev)) {
```

Comment:
The Doxygen doc for the functions gives the details on `cc110x_power_on()`. In short: It gives a pulse on the CS pin which will wake up the device (in case it was sleeping / off) and waits for the worst case power up time.

This needs to be done before the reset, as an command via SPI is used to reset the device. Unless the device is fully powered up, the SPI interface should not be used according to the data sheet. I'll add a comment to the code.

Reply:
What I meant was, even if the device has no `CPUID`, the `luid` module will set a value that relates to ``LUID_BACKUP_SEED`` [here](https://github.com/RIOT-OS/RIOT/blob/master/sys/luid/luid.c#L51). But anyway. That should also prevent the `while` to be executed more that once

Reply:
But there is no guarantee that [`luid_get()`](http://api.riot-os.org/group__sys__luid.html#gaf8a68544e02887426ca9b1e6d3edd3e2) will not return 0. For other cases, `0` could be a totally valid ID. But in the context of the `cc110x`, it is the broadcast address.

The API doc indicates that two subsequent returns values should differ, so I could replace the loop with an `if`. But the code is shorter this way and this code is safer, as with the `while` it is guaranteed to still generated valid L2 addresses even if the `luid_get()` semantics will ever changed in the future.

Reply:
I don't think `luid_get()` will change in that regard. It was primarily introduced, to allow for address allocation over multiple interfaces. Having the address be configured at compile time makes it nasty to flash multiple nodes. This method should only be used if there is no other source for the address.

Reply:
> This method should only be used if there is no other source for the address.

With the default parameters `luid_get()` is used, so configuration at compile time is not required when less than 20 nodes are used. (With 20 nodes the changes of an address conflict is already more than 50% due to the birthday paradox.)

> I don't think `luid_get()` will change in that regard.

Agreed. Still, the while is three lines less of code. So I'd still keep it for that reason.